src/d/actor/d_a_player_battle.inc

```diff
@@ -11,88 +11,513 @@
  */
 
 #include "d/actor/d_a_player_main.h"
+#include "d/actor/d_a_player_HIO.h"
 
 /* 8013E250-8013E384       .text setSpecialBattle__9daPy_lk_cFi */
-void daPy_lk_c::setSpecialBattle(int) {
-    /* Nonmatching */
+void daPy_lk_c::setSpecialBattle(int param_0) {
+    /* Nonmatching - regalloc */
+    float fVar1;
+    fopEn_enemy_c *pfVar3;
+    float dVar3;
+    float dVar4;
+  
+    if (((((param_0 == 0) || (!strcmp(dComIfGp_getStartStageName(), "GTower"))) &&
+          (mpAttnEntryA != NULL)) &&
+         ((((mpAttnEntryA->mType == 2 &&
+             (pfVar3 = (fopEn_enemy_c *)mpAttnActorLockOn, pfVar3 != NULL)) &&
+            ((pfVar3->group == 2 &&
+              ((!checkPlayerGuard() && (daPy_dmEcallBack_c::m_type != 1)))))) &&
+           (mEquipItem == daPyItem_SWORD_e)))) &&
+        (((pfVar3->mBtAttackType != 0 &&
+           (pfVar3->mBtStartFrame - daPy_HIO_battle_c0::m.field_0x4 <=
+            pfVar3->mBtNowFrame)) &&
+          (fVar1 < pfVar3->mBtEndFrame)))) {
```

Comment:
This just looks like Ghidra output and should be cleaned up, specifically the assignment -> comma operator -> condition sequences like `(pfVar3 = (fopEn_enemy_c *)mpAttnActorLockOn, pfVar3 != NULL)` should be split into separate if statements. Once everything's split into multiple statements instead of a single huge if statement a lot of the excessive parentheses and weird indentation should be fixable.

By the way mBtAttackType/mBtNowFrame etc all have inline getters in fopEn_enemy_c you can use here